src/formats/gtdb.rs

```diff
@@ -0,0 +1,134 @@
+use crate::rank::TaxRank;
+
+/// Read GTDB format into a Taxonomy object out of a `reader`.
+///
+/// Still somewhat experimental and may not support all GTDB features.
+pub fn load<R: Read>(reader: &mut R) -> TaxonomyResult<GeneralTaxonomy> {
+    let mut tax_names = Vec::new();
+    // Maps tax name to (index, lineage prefix)
+    let mut tax_names_idx: HashMap<String, (InternalIndex, String)> = HashMap::new();
+    let mut parent_ids = Vec::new();
+    let mut tax_ranks = Vec::new();
+
+    for (row_idx, row_result) in BufReader::new(reader).lines().enumerate() {
+        let line_num = row_idx + 1;
+        let row = row_result?;
+        // parts[0] -> accession number
+        // parts[1] -> lineage, `;` separated
+        let parts: Vec<_> = row.split('\t').collect();
+        if parts.len() != 2 {
+            return Err(Error::new(ErrorKind::ImportError {
+                line: line_num,
+                msg: "Expected tab-delimited line with exactly two parts (accession and lineage)"
+                    .to_owned(),
+            }));
+        }
+        let lineage: Vec<_> = parts[1].split(';').collect();
+
+        for (i, level) in lineage.iter().enumerate() {
+            let curr_prefix = lineage[..i].join(";");
+            match tax_names_idx.get(*level) {
+                Some((_, prev_prefix)) => {
+                    if curr_prefix != *prev_prefix {
+                        return Err(Error::new(ErrorKind::ImportError {
+                            line: line_num,
+                            msg: format!(
+                                "Inconsistent lineages for taxon {}: {} != {}",
+                                level, curr_prefix, prev_prefix
+                            ),
+                        }));
+                    }
+                    continue;
+                }
+                None => {
+                    let tax_rank = match &level[..3] {
+                        "d__" => TaxRank::Domain,
+                        "p__" => TaxRank::Phylum,
+                        "c__" => TaxRank::Class,
+                        "o__" => TaxRank::Order,
+                        "f__" => TaxRank::Family,
+                        "g__" => TaxRank::Genus,
+                        "s__" => TaxRank::Species,
+                        _ => TaxRank::Unspecified,
+                    };
+                    tax_names.push(level.to_string());
+                    tax_ranks.push(tax_rank);
+                    let idx = tax_names.len() - 1;
+                    tax_names_idx.insert(level.to_string(), (idx, curr_prefix));
+                    if i > 0 {
+                        let parent_id = tax_names_idx.get(lineage[i - 1]).unwrap().0;
+                        parent_ids.push(parent_id);
+                    } else {
+                        parent_ids.push(0);
+                    }
+                }
+            }
+        }
+    }
+
+    GeneralTaxonomy::from_arrays(
+        tax_names.clone(),
```

Comment:
i don't think you need to clone?

Reply:
If I remove the clone, I get this error because `tax_names` is used below (`Some(tax_names)`):

```
error[E0382]: use of moved value: `tax_names`
  --> src/formats/gtdb.rs:77:14
   |
12 |     let mut tax_names = Vec::new();
   |         ------------- move occurs because `tax_names` has type `Vec<std::string::String>`, which does not implement the `Copy` trait
...
75 |         tax_names,
   |         --------- value moved here
76 |         parent_ids,
77 |         Some(tax_names),
   |              ^^^^^^^^^ value used here after move
```

Reply:
oh I missed it was used just after, my bad.